lib/ansible/module_utils/iosxr.py

```diff
@@ -106,6 +106,11 @@ def load_config(module, commands, commit=False, replace=False, comment=None):
         module.fail_json(msg=err, commands=commands, rc=rc)
 
     rc, diff, err = exec_command(module, 'show commit changes diff')
+    if rc != 0:
```

Comment:
Problem is we set the task to 'changed' only if diff is not None in the run module function, but we may have changed the device even if the diff collected with 'show configuration' doesn't provide +/- changes.

I'm just putting the warning to let the user know that the diff may be bogus if old device but still have a consistent changed status.